analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -42,5 +42,20 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
         private bool IsDevelopmentCheck(SyntaxNode node, SemanticModel semanticModel) =>
             node is InvocationExpressionSyntax condition
             && IsValidationMethod(semanticModel, condition, condition.Expression.GetIdentifier()?.ValueText);
+
+        private static SyntaxNode EnclosingScope(SyntaxNode node) =>
+            node.Ancestors().FirstOrDefault(x => x.IsAnyKind(
+                SyntaxKind.AddAccessorDeclaration,
+                SyntaxKind.ConstructorDeclaration,
+                SyntaxKind.DestructorDeclaration,
+                SyntaxKind.GetAccessorDeclaration,
+                SyntaxKind.GlobalStatement,
+                SyntaxKindEx.InitAccessorDeclaration,
+                SyntaxKindEx.LocalFunctionStatement,
+                SyntaxKind.MethodDeclaration,
+                SyntaxKind.ParenthesizedLambdaExpression,
+                SyntaxKind.RemoveAccessorDeclaration,
+                SyntaxKind.SetAccessorDeclaration,
+                SyntaxKind.SimpleLambdaExpression));
```

Comment:
`IsAnyKind` allocates an array on each invocation and should not be used anymore (see e.g. https://github.com/SonarSource/sonar-dotnet/issues/8103)
```suggestion
        private static SyntaxNode EnclosingScope(SyntaxNode node) =>
            node.Ancestors().FirstOrDefault(x => x.Kind() is SyntaxKind.AddAccessorDeclaration
                or SyntaxKind.ConstructorDeclaration
                or SyntaxKind.DestructorDeclaration
                or SyntaxKind.GetAccessorDeclaration
                or SyntaxKind.GlobalStatement
                or SyntaxKindEx.InitAccessorDeclaration
                or SyntaxKindEx.LocalFunctionStatement
                or SyntaxKind.MethodDeclaration
                or SyntaxKind.ParenthesizedLambdaExpression
                or SyntaxKind.RemoveAccessorDeclaration
                or SyntaxKind.SetAccessorDeclaration
                or SyntaxKind.SimpleLambdaExpression));
```